src/sagas/nanoContract.js

```diff
@@ -440,7 +450,10 @@ export function* requestBlueprintInfo({ payload }) {
 export function* saga() {
   yield all([
     debounce(500, [[types.START_WALLET_SUCCESS, types.NANOCONTRACT_INIT]], init),
-    takeEvery(types.NANOCONTRACT_REGISTER_REQUEST, registerNanoContract),
+    takeEvery(
+      types.NANOCONTRACT_REGISTER_REQUEST,
+      safeEffect(registerNanoContract, registerNanoContractOnError)
```

Comment:
:star: Nice solution! For the future, I'm thinking of a way to make it more generic, and not require two handler functions for each effect.

Reply:
It would be awesome!